src/3d/symbols/qgsmesh3dsymbol.h

```diff
@@ -36,6 +37,33 @@
 class _3D_EXPORT QgsMesh3DSymbol : public QgsAbstract3DSymbol
 {
   public:
+
+    /**
+     * How to render the mesh
+     *
+     * \since QGIS 3.12
+     */
+    enum RendererType
+    {
+      //! Render the mesh as a simple 3D symbol
+      simpleSymbology = 0,
+      //! Render the mesh as advanced symbology (smoothed triangled, wireframe, contours...)
+      advancedSymbology
+    };
+
+    /**
+     * How to render the color of the mesh with advanced symbology
+     *
+     * \since QGIS 3.12
+     */
+    enum TextureType
+    {
+      //! Render the mesh with a unique color
+      uniqueColor = 0,
+      //! Render the mesh with a color ramp
+      colorRamp
+    };
```

Comment:
Can we have upper case starting letters?

And maybe rename TextureType to something else? Usage of texture is just an implementation detail... How about "RenderingStyle" ?